### PR TITLE
sync: use lib/log and add optional sentry reporting

### PR DESCRIPTION
**Pull request by bobheadxi**
This action has been failing for a while with no visibility - this PR adds error reporting on the sync job. I'll set up a Sentry DSN after the issue has been fixed ([thread](https://sourcegraph.slack.com/archives/C01CSS3TC75/p1660859630365779?thread_ts=1660672690.101049&cid=C01CSS3TC75))

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -2,15 +2,35 @@ module github.com/sourcegraph/clabot-config
...
+	github.com/kr/text v0.2.0 // indirect
+	github.com/mattn/go-colorable v0.1.12 // indirect
+	github.com/mattn/go-isatty v0.0.14 // indirect
+	github.com/pkg/errors v0.9.1 // indirect
+	github.com/rogpeppe/go-internal v1.8.1 // indirect
 	go.opencensus.io v0.23.0 // indirect
+	go.uber.org/atomic v1.7.0 // indirect
+	go.uber.org/multierr v1.6.0 // indirect
+	go.uber.org/zap v1.21.0 // indirect
 	golang.org/x/net v0.0.0-20220607020251-c690dde0001d // indirect
```
Comment (bobheadxi) [inline go.mod:34]: So it looks like `github.com/sourcegraph/log` is a rather sizeable dependency

File: tools/sync/main.go [hunk 2]
```diff
@@ -20,38 +20,51 @@ func main() {
 
 	ctx := context.Background()
 
+	sync := initLogging()
+	defer sync()
```
Comment (mucles) [inline tools/sync/main.go:24]: @bobheadxi,  It is bit unclear / confusing to me 😅  on why the initialization of the logger being deferred to execute at the end ?

Comment (burmudar) [reply tools/sync/main.go:24]: @bobheadxi correct me if I'm wrong - the sync here is to make **sure** anything left on the background threads or still in memory buffers is flushed/persisted if or when the program exits. Think of it like a cleanup of some sorts

File: tools/sync/main.go [hunk 3]
```diff
@@ -20,38 +20,51 @@ func main() {
...
+	logger.Info("listed responses", log.Int("responses", len(resps)))
+
+	// For each response, if not yet in config, add it
+	var added int
 	for _, resp := range resps {
-		// If not yet in config, add it
 		if _, exists := existingHandles[resp.GitHubHandle]; !exists {
-			fmt.Printf("Adding contributor %q (%s)\n", resp.GitHubHandle, resp.Name)
+			logger.Info("adding contributor",
+				log.String("gitHubHandle", resp.GitHubHandle),
```
Comment (burmudar) [inline tools/sync/main.go:53]:
```suggestion
				log.String("githubHandle", resp.GithubHandle),
```

**Review by burmudar — APPROVED** [review]

File: tools/sync/main.go (hunk 2 above)
Comment (mucles) [reply tools/sync/main.go:24]: @burmudar, yeah, the flush part makes sense to me why that is being deferred .
I guess confusion comes from trying to figure out where the sentry client is initialize... I was looking at the sentry doc example was trying to figure out where this step is happening :

```
err := sentry.Init(sentry.ClientOptions{
		Dsn: "https://examplePublicKey@o0.ingest.sentry.io/0",
	})
```
https://docs.sentry.io/platforms/go/

Is this where that sentry client is initiated: https://github.com/sourcegraph/clabot-config/pull/90/files#diff-5462a0d1d79a930e750a02cb1977089cb85ac9c80ff010b7457f87381632624bR12-R19 ?

**Review by mucles — APPROVED** [review]
Aside from my comment above ( not blocking just trying to learn and understand) this LGTM 😄

File: tools/sync/main.go (hunk 3 above)
Comment (bobheadxi) [reply tools/sync/main.go:53]: It's spelled GitHub I think 😛 

image

File: tools/sync/main.go (hunk 2 above)
Comment (bobheadxi) [reply tools/sync/main.go:24]: The Sentry client is initialized within the logging library :)

Comment (bobheadxi) [reply tools/sync/main.go:24]: It goes:

1. Add a log sink
2. Logging library takes sink configuration and does the necessary things to send error log entries to Sentry

File: tools/sync/main.go (hunk 3 above)
Comment (burmudar) [reply tools/sync/main.go:53]: ROFL 🤣 Don't mind me

File: tools/sync/main.go (hunk 2 above)
Comment (mucles) [reply tools/sync/main.go:24]: @bobheadxi  got it yeah that makes sense thank you 😄

**Pull request merged**